File: src/test/java/js/WebdriverIO/test/pageobjects/PlayPro/login.page.js
const { $ } = require('@wdio/globals');
const Page = require('./page');

/**
 * sub page containing specific selectors and methods for a specific page
 */
class LoginPage extends Page {
    /**
     * define selectors using getter methods
     */
    get inputEmail() {
        return $('input[name="email"]');
    }

    get inputPassword() {
        return $('input[name="password"]');
    }

    get btnSubmit() {
        return $('button[type="submit"]');
    }

    /**
     * a method to encapsule automation code to interact with the page
     * e.g. to login using username and password
     */
    async login(email, pwd) {
        await this.inputEmail.setValue(email);
        await this.inputPassword.setValue(pwd);
        await this.btnSubmit.click();
    }

    /**
     * overwrite specific options to adapt it to page object
     */
    open() {
        return super.open('connexion');
    }
}

module.exports = new LoginPage();

File: src/test/java/Selenide/UI/PlayPro/Web/Front/Pages/LoginPage.java
package Selenide.UI.PlayPro.Web.Front.Pages;

import com.codeborne.selenide.SelenideElement;

import static com.codeborne.selenide.Selenide.$;
import static com.codeborne.selenide.Selenide.open;

public class LoginPage {
    // Define web elements using locators
    SelenideElement emailInput = $("input[name=email]");
    SelenideElement passwordInput = $("input[name=password]");
    SelenideElement loginButton = $("button[type=submit]");

    // OR Use Classic Page Object
    // @FindBy(how = How.NAME, using = "q")
    //  private SelenideElement searchBox;

    // Method to open the login page
    public void openLoginPage() {
        open("https://demotenant.playpro.fr/connexion");
    }

    // Method to set email
    public void setEmail(String email) {
        emailInput.setValue(email);
    }

    // Method to set password
    public void setPassword(String password) {
        passwordInput.setValue(password);
    }

    // Method to click the login button
    public void clickLoginButton() {
        loginButton.click();
    }

}
